prov/verbs/src/windows/fi_verbs_nd.h

```diff
@@ -0,0 +1,259 @@
+	IND2Listener *listener;
+	ofi_atomic32_t cm_events_pending;
+};
+
+struct rdma_event_channel_vrb_nd {
+	struct rdma_event_channel channel;
+	struct dlistfd_head q;
+};
+
+struct rdma_cm_event_vrb_nd {
+	struct rdma_cm_event event;
+	struct dlist_entry entry;
+};
+
+struct ibv_cq_vrb_nd {
+	struct ibv_cq cq;
+	HANDLE ov_file;
+	IND2CompletionQueue *nd2cq;
+};
+
+struct ibv_mr_vrb_nd {
+	struct ibv_mr mr;
+	HANDLE ov_file;
+	IND2MemoryRegion *region;
+};
+
+struct ibv_qp_vrb_nd {
+	struct ibv_qp qp;
+	IND2QueuePair *nd2qp;
+	uint32_t max_inline_data;
+};
+
+struct ibv_comp_channel_vrb_nd {
+	struct ibv_comp_channel channel;
+	uint32_t num_pending;
+	ofi_mutex_t q_lock;
+	struct dlistfd_head q;
+};
+
+struct ibv_comp_event_vrb_nd {
+	struct ibv_cq *cq;
+	void *cq_ctx;
+	struct dlist_entry entry;
+};
+
+typedef void (*vrb_nd_event_t)(struct vrb_nd_event_base *base, DWORD bytes);
+typedef void (*vrb_nd_error_t)(struct vrb_nd_event_base *base, DWORD bytes,
+			       DWORD error);
+
+typedef struct vrb_nd_event_base {
+	OVERLAPPED ov;
+	vrb_nd_event_t event_cb;
+	vrb_nd_error_t error_cb;
+} vrb_nd_event_base;
+
+void CALLBACK vrb_nd_io_cb(DWORD error, DWORD bytes, LPOVERLAPPED ov);
+
+// Used in conjunction with IND2Listener::GetConnectionRequest()
+typedef struct vrb_nd_cm_listen_event {
+	vrb_nd_event_base base;
+	IND2Connector *connector;
+	struct rdma_cm_id *listen_id;
+} vrb_nd_cm_listen_event;
+
+void vrb_nd_cm_connreq_event(vrb_nd_event_base *base, DWORD bytes);
+void vrb_nd_cm_connreq_error(vrb_nd_event_base *base, DWORD bytes, DWORD error);
+
+// Used in conjunction with
+//   IND2Connector::Connect()
+//   IND2Connector::Accept()
+//   IND2Connector::CompleteConnect()
+//   IND2Connector::Disconnect()
+//   IND2Connector::NotifyDisconnect()
+typedef enum vrb_nd_cm_state {
+	CONNECT,
+	ACCEPT,
+	COMPLETE,
+	DISCONNECT,
+	DISCONNECTED
+} vrb_nd_cm_state;
+
+typedef struct vrb_nd_cm_connect_event {
+	vrb_nd_event_base base;
+	vrb_nd_cm_state state;
+	IND2Connector *connector;
+	struct rdma_cm_id *id;
+	struct rdma_event_channel *channel;
+	IND2QueuePair *qp;
+	struct {
+		const void *private_data;
+		uint8_t private_data_len;
+		uint8_t responder_resources;
+		uint8_t initiator_depth;
+	} param;
+} vrb_nd_cm_connect_event;
+
+void vrb_nd_get_connection_data(IND2Connector *connector,
+				struct rdma_cm_event_vrb_nd *entry_nd);
+
+void vrb_nd_cm_connect_ack(vrb_nd_event_base *base, DWORD bytes);
+void vrb_nd_cm_connect_nack(vrb_nd_event_base *base, DWORD bytes, DWORD error);
+
+// Used in conjunction with IND2CompletionQueue::Notify
+typedef struct vrb_nd_cq_notification {
+	vrb_nd_event_base base;
+	struct ibv_comp_channel_vrb_nd *ch_nd;
+	struct ibv_cq *cq;
+} vrb_nd_cq_notification;
+
+void vrb_nd_cq_notify_event(vrb_nd_event_base *base, DWORD bytes);
+void vrb_nd_cq_notify_error(vrb_nd_event_base *base, DWORD bytes, DWORD error);
+
+inline void trace(const char *function)
+{
+	VRB_INFO(FI_LOG_FABRIC, "%s\n", function);
+}
```

Comment:
Use VRB_INFO directly, or VRB_TRACE (if that exists) since this is tracing a function call.  The logging macros automatically log the function name, so calling the FI_LOG macro with an empty string is sufficient.